### ISSUE TITLE
Support ARKit/ARCore camera systems

### ISSUE THREAD

**Issue by Foxman13** [microsoft/MixedRealityToolkit-Unity#3616]
## Overview
When building a solution with a scene that is shared between platforms, the MRTK camera system overrides the camera settings for the mobile platform.

## Expected Behavior
MRTK detects the platform and configures the camera system accordingly.

## Actual Behavior
MRTK defaults to the HoloLens camera system.

## Steps to reproduce
Create a scene and attempt to run on mobile device such as ARKit. The camera settings will be wrong and the scene will not render correctly.

## Unity Editor Version
2018.3.*

## Mixed Reality Toolkit Release Version
MRTK v2

**Comment by davidkline-ms** [comment]
@Foxman13. Thanks for the feedback! ARKit and ARCore are on the team's roadmap. More information should be available in the coming weeks.

If you have any specific examples of the camera configuration differences, please feel free to add them to this issue thread.

**Comment by chrisfromwork** [comment]
I am building some initial work in spectator view to use arkit and arcore cameras (through ar foundation). I may as well add this as a blocking item for getting spectator view into the MRTK.

**Comment by benjamincoven** [comment]
Hello, we have experienced using ARCore on an existing HTK project, we had some minor troubles but we could make it works.

First, we have added the ARCoreSession component on the MixedRealityCameraParent and the ARCoreBackgroundRenderer on the MixedRealityCamera. Do not add a TrackedPoseDriver on the MixedRealityCamera.

Secondly, create some code with an Awake that will:

- Detects if you are in Android or UWP

- Enable the MixedRealityCameraManager if you are in UWP and disable if you are on Android

- Enable ARCoreSession and ARCoreBackgroundRenderer if you are in Android and disable if you are in UWP

- If you are on Android add the component TrackedPoseDriver (still in the Awake)
```
TrackedPoseDriver tPD = trackedPoseDriver.AddComponent<TrackedPoseDriver>();
            tPD.SetPoseSource(TrackedPoseDriver.DeviceType.GenericXRDevice, TrackedPoseDriver.TrackedPose.ColorCamera);
            tPD.trackingType = TrackedPoseDriver.TrackingType.RotationAndPosition;
            tPD.UseRelativeTransform = true;
            tPD.updateType = TrackedPoseDriver.UpdateType.Update;
```

If you have any tips and or feedback for ARFundation that would be nice. We have not tried this tech and using it instead of ARCore could be better.
